agent/hcp/telemetry_provider.go

```diff
@@ -43,57 +44,54 @@ type hcpProviderImpl struct {
 // dynamicConfig is a set of configurable settings for metrics collection, processing and export.
 // fields MUST be exported to compute hash for equals method.
 type dynamicConfig struct {
-	Endpoint *url.URL
-	Labels   map[string]string
-	Filters  *regexp.Regexp
+	disabled bool
+	endpoint *url.URL
+	labels   map[string]string
+	filters  *regexp.Regexp
 	// refreshInterval controls the interval at which configuration is fetched from HCP to refresh config.
-	RefreshInterval time.Duration
+	refreshInterval time.Duration
 }
 
-// NewHCPProvider initializes and starts a HCP Telemetry provider with provided params.
-func NewHCPProvider(ctx context.Context, hcpClient client.Client, telemetryCfg *client.TelemetryConfig) (*hcpProviderImpl, error) {
-	refreshInterval := telemetryCfg.RefreshConfig.RefreshInterval
-	// refreshInterval must be greater than 0, otherwise time.Ticker panics.
-	if refreshInterval <= 0 {
-		return nil, fmt.Errorf("invalid refresh interval: %d", refreshInterval)
-	}
-
-	cfg := &dynamicConfig{
-		Endpoint:        telemetryCfg.MetricsConfig.Endpoint,
-		Labels:          telemetryCfg.MetricsConfig.Labels,
-		Filters:         telemetryCfg.MetricsConfig.Filters,
-		RefreshInterval: refreshInterval,
-	}
-
-	t := &hcpProviderImpl{
-		cfg:       cfg,
+// NewHCPProvider initializes and starts a HCP Telemetry provider.
+func NewHCPProvider(ctx context.Context, hcpClient client.Client) *hcpProviderImpl {
+	h := &hcpProviderImpl{
+		// Initialize with default config values.
+		cfg: &dynamicConfig{
+			labels:          map[string]string{},
+			filters:         client.DefaultMetricFilters,
+			refreshInterval: defaultTelemetryConfigRefreshInterval,
+			endpoint:        nil,
+			disabled:        true,
+		},
 		hcpClient: hcpClient,
 	}
 
-	go t.run(ctx, refreshInterval)
+	go h.run(ctx)
 
-	return t, nil
+	return h
 }
 
 // run continously checks for updates to the telemetry configuration by making a request to HCP.
-func (h *hcpProviderImpl) run(ctx context.Context, refreshInterval time.Duration) {
-	ticker := time.NewTicker(refreshInterval)
+func (h *hcpProviderImpl) run(ctx context.Context) {
+	// Try to initialize config once before starting periodic fetch.
+	h.updateConfig(ctx)
+
+	ticker := time.NewTicker(h.cfg.refreshInterval)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
-			if newCfg := h.getUpdate(ctx); newCfg != nil {
-				ticker.Reset(newCfg.RefreshInterval)
+			if newRefreshInterval := h.updateConfig(ctx); newRefreshInterval > 0 {
```

Comment:
Missed this in previous PR, but we can really just return the refreshInterval to reset the ticker instead of the whole config.